RecoHI/HiTracking/plugins/HIPixelTracksSlimmer_miniAOD.cc

```diff
@@ -0,0 +1,256 @@
+void HIPixelTracksSlimmer_miniAOD::beginStream(edm::StreamID) {}
+
+// ------------ method called once each stream after processing all runs, lumis and events  ------------
+void HIPixelTracksSlimmer_miniAOD::endStream() {}
+
+// ------------ method called when starting to processes a run  ------------
+/*
+void
+HIPixelTracksSlimmer_miniAOD::beginRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a run  ------------
+/*
+void
+HIPixelTracksSlimmer_miniAOD::endRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void
+HIPixelTracksSlimmer_miniAOD::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void
+HIPixelTracksSlimmer_miniAOD::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+void HIPixelTracksSlimmer_miniAOD::addPackedCandidate(std::vector<pat::PackedCandidate>& cands,
+                                                      const reco::TrackRef& trk,
+                                                      const reco::VertexRef& pvSlimmed,
+                                                      const reco::VertexRefProd& pvSlimmedColl,
+                                                      bool passPixelTrackSel) const {
+  const float mass = 0.13957018;
+
+  int id = 211 * trk->charge();
+
+  reco::Candidate::PolarLorentzVector p4(trk->pt(), trk->eta(), trk->phi(), mass);
+  cands.emplace_back(
+      pat::PackedCandidate(p4, trk->vertex(), trk->pt(), trk->eta(), trk->phi(), id, pvSlimmedColl, pvSlimmed.key()));
+
+  if (passPixelTrackSel)
+    cands.back().setTrackHighPurity(true);
+  else
+    cands.back().setTrackHighPurity(false);
+
+  cands.back().setTrackProperties(*trk, covarianceSchema_, covarianceVersion_);
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void HIPixelTracksSlimmer_miniAOD::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("srcTracks", edm::InputTag("hiConformalPixelTracks"));
+  desc.add<edm::InputTag>("srcVertices", edm::InputTag("offlinePrimaryVertices"));
+  desc.add<edm::InputTag>("srcPrimaryVertices", edm::InputTag("offlineSlimmedPrimaryVertices"));
+  desc.add<double>("dzSigCut", double(100.0));
+  desc.add<double>("dxySigCut", double(100.0));
+  desc.add<double>("dzSigHP", double(100.0));
+  desc.add<double>("dxySigHP", double(100.0));
+  desc.add<double>("ptMax", double(1.0));
+  desc.add<int>("covarianceVersion", int(1));
+  desc.add<int>("covarianceSchema", int(0));
```

Comment:
types are not needed on rhs, certainly not for basic types
```suggestion
  desc.add<edm::InputTag>("srcTracks", {"hiConformalPixelTracks"});
  desc.add<edm::InputTag>("srcVertices", {"offlinePrimaryVertices"});
  desc.add<edm::InputTag>("srcPrimaryVertices", {"offlineSlimmedPrimaryVertices"});
  desc.add<double>("dzSigCut", 100.0);
  desc.add<double>("dxySigCut", 100.0);
  desc.add<double>("dzSigHP", 100.0);
  desc.add<double>("dxySigHP", 100.0);
  desc.add<double>("ptMax", 1.0);
  desc.add<int>("covarianceVersion", 1);
  desc.add<int>("covarianceSchema", 0);
```

Reply:
BTW, why not set all parameters already matching the needs in hipixelTracks ?

Reply:
This comment is included in this commit: https://github.com/cms-sw/cmssw/pull/30313/commits/deebf6766c730a8c97ac3f3fe90b065b24d5e70d